service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt

```diff
@@ -78,120 +89,163 @@ private fun Database.Read.getUnitOperationalDays(unitId: DaycareId, dateRange: F
+        to_char((res.end_time AT TIME ZONE 'Europe/Helsinki')::time, 'HH24:MI') AS reservation_end_time,
+        att.arrived AS attendance_start,
+        att.departed AS attendance_end,
+        ab.absence_type
+    FROM generate_series(:start, :end, '1 day') t
+    JOIN placement pl ON pl.unit_id = :unitId AND daterange(pl.start_date, pl.end_date, '[]') @> t::date
+    JOIN person p ON p.id = pl.child_id
+    LEFT JOIN daycare_group_placement dgp on dgp.daycare_placement_id = pl.id AND daterange(dgp.start_date, dgp.end_date, '[]') @> t::date
+    LEFT JOIN daycare_group dg ON dg.id = dgp.daycare_group_id
+    LEFT JOIN attendance_reservation res ON res.child_id = p.id AND res.start_date = t::date
+    LEFT JOIN child_attendance att ON att.child_id = p.id AND (att.arrived AT TIME ZONE 'Europe/Helsinki')::date = t::date
+    LEFT JOIN absence ab ON ab.child_id = p.id AND ab.date = t::date
     """.trimIndent()
 )
     .bind("unitId", unitId)
-    .bind("dateRange", dateRange)
+    .bind("start", dateRange.start)
+    .bind("end", dateRange.end)
     .mapTo<UnitAttendanceReservations.QueryRow>()
     .toList()
 
+// currently queried separately to be able to use existing mapper
+private fun Database.Read.getDailyServiceTimes(childIds: Set<UUID>) = createQuery(
+    """
+    SELECT *
+    FROM daily_service_time
+    WHERE child_id = ANY(:childIds)
+    """.trimIndent()
+)
+    .bind("childIds", childIds.toTypedArray())
+    .map { row -> toDailyServiceTimes(row) }
+    .toList()
+    .filterNotNull()
+    .associateBy { it.childId }
+
+private fun mapChildReservations(rows: List<UnitAttendanceReservations.QueryRow>, serviceTimes: Map<UUID, DailyServiceTimes>): List<UnitAttendanceReservations.ChildReservations> {
+    return rows
+        .groupBy { it.child }
+        .map { (child, rowsByChild) ->
+            UnitAttendanceReservations.ChildReservations(
+                child = child.copy(dailyServiceTimes = serviceTimes.get(child.id)),
+                dailyData = rowsByChild
+                    .associateBy { it.date }
+                    .map { (date, rowByDateAndChild) ->
+                        UnitAttendanceReservations.DailyChildData(
+                            date = date,
+                            reservation = rowByDateAndChild.reservation?.let {
+                                UnitAttendanceReservations.ReservationTimes(
+                                    startTime = it.startTime,
+                                    endTime = it.endTime
+                                )
+                            },
+                            attendance = rowByDateAndChild.attendance?.let {
+                                UnitAttendanceReservations.AttendanceTimes(
+                                    startTime = it.start.toLocalTime().format(DateTimeFormatter.ofPattern("HH:mm")),
+                                    endTime = it.end?.toLocalTime()?.format(DateTimeFormatter.ofPattern("HH:mm")),
```

Comment:
For consistency either do this formatting in the database query like with reservations or convert the reservations into local times in code.